src/Cms/Auth.php

```diff
@@ -523,6 +506,25 @@ public function validatePassword(string $email, string $password)
             throw new PermissionException($message);
         }
 
+        return $email;
+    }
+
+    /**
+     * Validates the user credentials and returns the user object on success;
+     * otherwise logs the failed attempt
+     *
+     * @param string $email
+     * @param string $password
+     * @return \Kirby\Cms\User
+     *
+     * @throws \Kirby\Exception\PermissionException If the rate limit was exceeded or if any other error occurred with debug mode off
+     * @throws \Kirby\Exception\NotFoundException If the email was invalid
+     * @throws \Kirby\Exception\InvalidArgumentException If the password is not valid (via `$user->login()`)
```

Comment:
It's not very important, but while you're in it; 
Do we need this? As standard, we only add Exceptions thrown from within the method to the doc blocks. 

For example, same for `createChallenge()` method and `PermissionException`..

Reply:
Ideally all exceptions should be documented, no matter where in the chain they are thrown. After all the user of this method doesn't know which methods are called internally.

The only issue so far has been that it's incredibly hard to find all exceptions and keep them up to date. That's why we generally don't do it at the moment, but I wouldn't remove already existing hints.